src/contexts/global.tsx

```diff
@@ -116,19 +122,33 @@ export const GlobalContextProvider: FC<{ overrideContextValue?: PartialDeep<Glob
   }
 
   const lockWallet = () => {
+    setPassphraseDoubleHashed('')
     setCurrentAccountName('')
     setWallet(undefined)
   }
 
-  const login = async (accountName: string, password: string, callback: () => void) => {
+  const login = async (accountName: string, password: string, callback: () => void, passphrase?: string) => {
     const walletEncrypted = Storage.load(accountName)
     if (!walletEncrypted) {
       setSnackbarMessage({ text: 'Unknown wallet name', type: 'alert' })
       return
     }
     try {
-      const wallet = walletOpen(password, walletEncrypted)
+      let wallet = walletOpen(password, walletEncrypted)
       if (!wallet) return
+      if (passphrase) {
+        wallet = getWalletFromMnemonic(`${wallet.mnemonic} ${passphrase}`)
```

Comment:
This is not how passphrase is used in BIP39. We'd better test against those test vectors.

bip39 supports passphrase already, but it's named as `password`.

Reply:
> To create a binary seed from the mnemonic, we use the PBKDF2 function with a mnemonic sentence (in UTF-8 NFKD) used as the password and the string "mnemonic" + passphrase (again in UTF-8 NFKD) used as the salt.

`passphrase` should be salt, instead of being the actual 13th/25th world

Reply:
I'm going to find some implementations to verify, but I'm pretty sure then all it means is I should remove the space between the mnemonic and the passphrase, example:

* Current: "hello world some word extraword"
* Corrected: "hello wold some wordextraword"

salts are appended to hash inputs i.e. `h(input + salt)` so this should be right... Going to verify though!

Reply:
This should change probably be put into `alephium/sdk` instead - I'll go do it there and change the function call here

Reply:
Fixed in alephium/sdk and applied here